docs/af5-app-from-scratch/modules/ROOT/pages/implement-command-subscribe-student.adoc

```diff
@@ -0,0 +1,601 @@
+                        .hasMessage("Student subscribed to too many courses")
+                );
+    }
+
+}
+----
+
+These tests demonstrate the behavior we want to implement, checking both successful and error cases.
+To those tests we need to implement the command handler for `SubscribeStudent` command.
+As you've seen before, for the behavior which is based on some state (so we have something in a Given phase of the test), we need to have State for our command handler to validate commands against it.
+Let's make it right away!
+
+[source,java]
+.src/main/java/io/axoniq/demo/university/faculty/write/subscribestudent/SubscribeStudent.java
+----
+package io.axoniq.demo.university.faculty.write.subscribestudent;
+
+class SubscribeStudentCommandHandler {
+
+    private static final int MAX_COURSES_PER_STUDENT = 3; // <1>
+
+    @CommandHandler
+    void handle(
+            SubscribeStudent command,
+            @InjectEntity State state, // <2>
+            EventSink eventSink,
+            ProcessingContext processingContext
+    ) {
+        var events = decide(command, state); // <3>
+        var messages = toMessages(events); // <4>
+        eventSink.publish(processingContext, messages); // <5>
+    }
+
+    private List<StudentSubscribed> decide(SubscribeStudent command, State state) {
+        // todo: check business rules
+        return List.of(new StudentSubscribed(command.studentId().raw(), command.courseId().raw()));
+    }
+
+    private static List<EventMessage<?>> toMessages(List<StudentSubscribed> events) {
+        return events.stream()
+                     .map(SubscribeStudentCommandHandler::toMessage)
+                     .collect(Collectors.toList());
+    }
+
+    private static EventMessage<?> toMessage(Object payload) {
+        return new GenericEventMessage<>(
+                new MessageType(payload.getClass()),
+                payload
+        );
+    }
+
+    @EventSourcedEntity // <6>
+    static class State {
+        // todo: define state necessary to accept or reject `SubscribeStudent` command
+    }
+
+}
+----
+
+<1> For the sample simplicity, we hardcoded the maximum number of courses per student.
+Each student can subscribe up to 3 courses at the same time.
+<2> We use `@InjectEntity` to inject the state object.
+<3> This is your domain model invocation.
+You may keep it in the command handler as on the example or make the function unaware of the infrastructure like Axon Framework.
+This functional resembles the `Decider` pattern.
+<4> As before, we translate events (which are a result of our domain function) to messages to be able to publish them.
+<5> We use `EventSink` to publish events.
+<6> Before we defined `tagKey` in `@EventSourcedEntity` annotation, now we cannot do that, because we require events about every subscription of a student and every subscription to the course, so we have multiple business concepts related to a business process.
```

Comment:
```suggestion
<6> Before we defined a `tagKey` in `@EventSourcedEntity` annotation. Now we cannot do that, because we require events about every subscription of a student and every subscription to the course. So we have multiple business concepts related to a business process!
```